src/components/marketplace/buy_sell.tsx

```diff
@@ -203,26 +251,52 @@ class BuySell extends React.Component<Props, State> {
                             <TokenText>wETH</TokenText>
                         </TokenContainer>
                     </FieldContainer>
-                    <Button theme="secondary" onClick={tab === Tab.Buy ? this.buy : this.sell}>
-                        {tab === Tab.Buy ? 'Buy' : 'Sell'} {selectedTokenSymbol}
+
+                    <LabelContainer>
+                        <Label>Order Details</Label>
+                        <InnerTabs tabs={ethUsdTabs} />
+                    </LabelContainer>
+                    <Row>
+                        <Label color={themeColors.textLight}>Fee</Label>
+                        <Value>
+                            {orderDetailType === OrderDetailsType.Usd
+                                ? `$ ${this.state.orderFeeDollar.toFixed(2)}`
+                                : `${this.state.orderFeeEther.toString()} Eth`}
+                        </Value>
+                    </Row>
+
+                    <Button theme="secondary" onClick={tab === OrderSide.Buy ? this.buy : this.sell}>
+                        {tab === OrderSide.Buy ? 'Buy' : 'Sell'} {selectedTokenSymbol}
                     </Button>
                 </Content>
             </BuySellWrapper>
         );
     };
 
-    public changeTab = (tab: Tab) => () => this.setState({ tab });
+    public changeTab = (tab: OrderSide) => () => this.setState({ tab }, () => this.updateOrderFee());
 
     public updateMakerAmount = (newValue: BigNumber) => {
-        this.setState({
-            makerAmount: newValue,
-        });
+        this.setState(
+            {
+                makerAmount: newValue,
+            },
+            () => this.updateOrderFee(),
+        );
     };
 
     public updatePrice: React.ReactEventHandler<HTMLInputElement> = e => {
         const price = parseFloat(e.currentTarget.value);
+        this.setState({ price }, () => this.updateOrderFee());
+    };
 
-        this.setState({ price });
+    public updateOrderFee = async () => {
+        const orderPrice = this.state.makerAmount.mul(this.state.price);
+        const orderFeeEther = orderDetailsFeeEther(orderPrice, this.state.tab);
+        const orderFeeDollar = await orderDetailsFeeDollar(orderPrice, this.state.tab);
```

Comment:
Does this mean that we are going to make a request to the CMC API every key stroke? We may hit a rate limit if we do that.

Two ways of fixing this come to mind: we can periodically ask for the price and save it in the store, or we can add some caching to the function so that the price is not asked more than once every X minutes.

The first option is probably better, but the second one is easier. We can make the cache thing for now and add the price to the store in another PR.

Reply:
More on this: manually calling `updateOrderFee` after `this.setState` doesn't look good. Since the fee is computed from the order values, I think it makes more sense to do it in the render function. But we can't do it there because this function is async. So more reason to just have the ETH price in the store. Again: no need to do it now, just something to keep in mind.

Reply:
I implemented a cache with interval, we could maybe create another issue to move all this things to the store directly

Reply:
I opened #77.